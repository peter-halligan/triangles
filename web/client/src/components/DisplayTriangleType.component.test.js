import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {DisplayTriangleType }from './DisplayTriangleType.component'

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<DisplayTriangleType type={"scalene"} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test(' renders scalene when it is passed as a prop ',  () =>{
  const tree = renderer
    .create(<DisplayTriangleType type={"scalene"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test(' renders isosceles when it is passed as a prop ',  () =>{
  const tree = renderer
    .create(<DisplayTriangleType type={"isosceles"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test(' renders equilateral when it is passed as a prop ',  () =>{
  const tree = renderer
    .create(<DisplayTriangleType type={"equilateral"} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
