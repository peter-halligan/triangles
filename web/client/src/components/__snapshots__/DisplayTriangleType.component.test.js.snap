// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders equilateral when it is passed as a prop  1`] = `
<p>
  equilateral
</p>
`;

exports[` renders isosceles when it is passed as a prop  1`] = `
<p>
  isosceles
</p>
`;

exports[` renders scalene when it is passed as a prop  1`] = `
<p>
  scalene
</p>
`;
